src/provable/base58.unit-test.ts

```diff
@@ -1,28 +1,26 @@
 import { fromBase58Check, toBase58Check } from './base58.js';
 import { Ledger, isReady, shutdown } from '../snarky.js';
 import { expect } from 'expect';
+import { test, Random } from '../lib/testing/property.js';
 
 await isReady;
 
-function check(bytes: number[], version: number) {
+let bytes = Random.bytes(Random.nat(100));
+let version = Random.nat(100);
+
+test(bytes, version, (bytes, version, assert) => {
   let binaryString = String.fromCharCode(...bytes);
   let ocamlBytes = { t: 9, c: binaryString, l: bytes.length };
   let base58Ocaml = Ledger.encoding.toBase58(ocamlBytes, version);
 
   // check consistency with OCaml result
   let base58 = toBase58Check(bytes, version);
-  expect(base58).toEqual(base58Ocaml);
+  assert(base58 === base58Ocaml, 'base58 agrees with ocaml');
 
   // check roundtrip
   let recoveredBytes = fromBase58Check(base58, version);
   expect(recoveredBytes).toEqual(bytes);
-}
-
-check([0, 1, 2, 3, 4, 5], 10);
-check([250, 200, 150, 100, 50, 0], 1);
-check([0, 0], 0);
-check([1], 100);
-check([], 0x01);
```

Comment:
Can you add optional argument to specify examples that must be tested? The Jane street ocaml Quickcheck library has this and it’s nice for these kinds of situations where you have specific test cases you want to include with 100% probability.

Reply:
good idea! I'll add a combinator that takes a generator and enhances it to create some constant values unconditionally, as the first outputs of `next()`

Reply:
done! https://github.com/o1-labs/snarkyjs/pull/718/commits/4082beab9f8eaac915cd76884e49683dc3eab636